File: Application/Dto/School/ClassSubjectSemesterDto.cs
using Domain.Entities_v2.Types;

namespace Application.Dto.School;

public class ClassSubjectSemesterDto: IBaseDto
{
    public int Id { get; set; }
    public DateTime Created { get; set; } 
    public DateTime Updated { get; set; }
    public Status Status { get; set; }
    
    public int Count { get; set; }
    public int ClassId { get; set; }
    public int SubjectId { get; set; }
    public int SemesterId { get; set; }
    
    public int? TeacherId { get; set; }
    public string TeacherName { get; set; } = "";
    public string SubjectName { get; set; } = "";
    public string ClassName { get; set; } = "";
    
    public List<ClassSubjectSemesterInstanceDto>? ClassSubjectSemesterInstances { get; set; } = null!;
}

File: frontend/src/Domain/Dtos/ClassSubjectSemesterDto.ts
import EntityBase from '@/domain/EntityBase.ts';
import CssInstanceDto from '@/domain/dtos/CssInstanceDto.ts';

interface ClassSubjectSemesterDto extends EntityBase {
  count: number;
  classId: number;
  subjectId: number;
  semesterId: number;

  teacherName?: string;
  subjectName?: string;
  className?: string;

  teacherId?: number;

  classSubjectSemesterInstances?: CssInstanceDto[];
}

export default ClassSubjectSemesterDto;
